test/ng/directive/ngEventDirsSpec.js

```diff
@@ -148,35 +148,131 @@ describe('event directives', function() {
       expect($rootScope.blur).toHaveBeenCalledOnce();
       expect(element.val()).toBe('newValue');
     }));
+  });
 
+  describe('events in digest phases', function() {
 
-    it('should throw the error when wrapped in an apply',
-        inject(function($rootScope, $compile) {
-      element = $compile('<button ng-click="click()">Click</button>')($rootScope);
+    it('should not stop the digest when the handler throws an error', function() {
 
-      $rootScope.click = function() {
-        throw new Error('listener error');
-      };
+      module(function($exceptionHandlerProvider) {
+        $exceptionHandlerProvider.mode('log');
+      });
+
+      inject(function($rootScope, $compile, $exceptionHandler, $log) {
+
+        element = $compile('<button ng-click="click()">Click</button>')($rootScope);
+        expect($log.assertEmpty());
+        $rootScope.click = function() {
+          throw new Error('listener error');
+        };
+
+        $rootScope.do = function() {
+          $log.log('done');
+        };
 
-      expect(function() {
         $rootScope.$apply(function() {
           element.triggerHandler('click');
         });
-      }).toThrowError('listener error');
-    }));
+        $rootScope.do();
```

Comment:
I was thinking this would be one line up, within the `$apply`. Or maybe it would be more clear if we did:
```
$rootScope.do = function() {
    element.triggerHandler('click');
    $log.log('done');
}
```
Then invoke `$rootScope.do` within vs  outside `$apply`. Those 2 cases should be consistent, both with `$exceptionHandler` throwing (the `done` never runs) vs only logging (the `done` always runs). I think this commit is required for those to be consistent?

Reply:
Ok, if we do 
```
      $rootScope.do = function() {
          element.triggerHandler('click');
          $log.log('done');
        };

        $rootScope.$apply(function() {
          $rootScope.do();
        });
```
then we need the try-catch. 
I guess the question is now for which behavior  it's more important to be consistent? Because the try catch changes the behavior when we throw inside a digest. I assume that behavior is not as important.

Reply:
Can you show an example of that change?